src/record.js

```diff
@@ -31,35 +52,36 @@ export class BagHeader extends Record {
   connectionCount: number;
   chunkCount: number;
 
-  constructor(fields: { [key: string]: Buffer }, buffer: Buffer) {
+  constructor(fields: { [key: string]: Uint8Array }, buffer: Uint8Array) {
     super(fields, buffer);
-    this.indexPosition = readUInt64LE(fields.index_pos);
-    this.connectionCount = fields.conn_count.readInt32LE(0);
-    this.chunkCount = fields.chunk_count.readInt32LE(0);
+    this.indexPosition = readUInt64LE(fields.index_pos, 0);
+    this.connectionCount = readInt32LE(fields.conn_count, 0);
+    this.chunkCount = readInt32LE(fields.chunk_count, 0);
   }
 }
 
 export class Chunk extends Record {
   static opcode = 5;
   compression: string;
   size: number;
-  data: Buffer;
+  data: Uint8Array;
 
-  constructor(fields: { [key: string]: Buffer }, buffer: Buffer) {
+  constructor(fields: { [key: string]: Uint8Array }, buffer: Uint8Array) {
     super(fields, buffer);
-    this.compression = fields.compression.toString();
-    this.size = fields.size.readUInt32LE(0);
+    this.compression = new TextDecoder().decode(fields.compression);
+    this.size = readUInt32LE(fields.size, 0);
     this.data = buffer;
   }
 }
 
-const getField = (fields: { [key: string]: Buffer }, key: string) => {
+const getField = (fields: { [key: string]: Uint8Array }, key: string) => {
   if (fields[key] === undefined) {
     throw new Error(`Connection header is missing ${key}.`);
   }
-  return fields[key].toString();
+  return new TextDecoder().decode(fields[key]);
```

Comment:
In `MessageReader.js` we [found](https://github.com/cruise-automation/rosbag.js/blob/master/src/MessageReader.js#L74) the constant overhead of `TextDecoder` to make it a fair bit slower than constructing the strings by repeated concatenation. This code is a bit performance-sensitive, worth checking that this doesn't add much to the running time.